tools/generator/CodeGenerationOptions.cs

```diff
@@ -156,6 +160,46 @@ string GetNullable (string s)
 			return NullableOperator;
 		}
 
+		// Encoding format:
+		// - Type name prefix: _JniMarshal_PP
+		// - Parameter types, using JNI encoding, e.g. Z is boolean, I is int, etc. Exception: Reference types, normally encoded as L…;, are instead just L.
+		// - Another _.
+		// - Return type, encoded as with parameters. A void return type is V.
+		internal string GetJniMarshalDelegate (Method method)
+		{
+			var sb = new StringBuilder ("_JniMarshal_PP");
+
+			foreach (var p in method.Parameters)
+				sb.Append (GetJniTypeCode (p.Symbol));
+
+			sb.Append ("_");
+
+			sb.Append (method.IsVoid ? "V" : GetJniTypeCode (method.RetVal.Symbol));
```

Comment:
Not sure the `.IsVoid` check is required; shouldn't `method.RetVal` *already* be an `ISymbol` instance for `void`, in which `symbol.JniName`==`V`?